File: exercism/scala/bob/src/main/scala/Bob.scala
object Bob {
  def response(statement: String): String = {
    if(statement.isBlank){
      "Fine. Be that way!"
    }
    else if(capital(statement) && statement.trim().endsWith("!")){
      "Whoa, chill out!"
    }else if(capital(statement) && statement.trim().endsWith("?")){
      "Calm down, I know what I'm doing!"
    }
    else if(capital(statement)){
      "Whoa, chill out!"
    }
    else if(statement.trim().endsWith("?")){
      "Sure."
    }else{
      "Whatever."
    }

  }

  def capital(statement: String): Boolean ={
    statement.filter(c => c.isLetter).forall(c => c.isUpper) && statement.exists(c => c.isLetter);
  }

}

File: exercism/python/bob/bob.py
def response(statement):
    if isblank(statement):
        return "Fine. Be that way!"
    if str.isupper(statement) and str.endswith(str.strip(statement), "!"):
        return "Whoa, chill out!"
    if str.isupper(statement) and str.endswith(str.strip(statement), "?"):
        return "Calm down, I know what I'm doing!"
    if str.isupper(statement):
        return "Whoa, chill out!"
    if str.endswith(str.strip(statement), "?"):
        return "Sure."
    return "Whatever."


def isblank(statement):
    return not (statement and statement.strip())